node/node.go

```diff
@@ -576,6 +578,12 @@ func (n *Node) Stop() error {
 	failure := &StopError{
 		Services: make(map[reflect.Type]error),
 	}
+	for kind, service := range n.subservices {
```

Comment:
It would be better if you leave a comment about the reason why we're doing this (as described in the description of this PR)

Reply:
Added comments. 3ec99a7a1786f3fb360d8a1071a7d9de6a325697